CI: Automatically updating the K8s version failed

### Which tests are failing?

```
updated version: [v1.30.2, v1.25.16, v1.26.15, v1.27.13, v1.28.9, v1.29.4, v1.23.17, v1.24.17, v1.22.17]
update failed, please check.
Error: Process completed with exit code 1.
```

If kind/node is updated to version 1.30.2 instead of 1.31.0 , the automatic update will fail.

### Job's Link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

--- a/.github/workflows/auto-diff-k8s-ci.yaml
+++ b/.github/workflows/auto-diff-k8s-ci.yaml
@@ -128,12 +128,13 @@ jobs:
       - name: get latest kind node version
         run: |
           bash ./test/scripts/update-kindnode-version.sh ${{ env.MINIMUM_K8S_VERSION }} ${{ env.K8S_MATRIX_FILE_PATH }}
-          if [ $? -eq 0 ]; then
+          if [ $? -ne 0 ]; then
             echo "./test/scripts/update-kindnode-version.sh script failed"
             echo "SKIP_CREATE_PR=true" >> $GITHUB_ENV
             exit 0
           fi
           git_status=$(git status --porcelain)
+          echo "git status: ${git_status}"
           if [[ -n "$git_status" ]]; then
             echo "SKIP_CREATE_PR=false" >> $GITHUB_ENV
             echo "The .github/workflows/auto-diff-k8s-ci.yaml of Spiderpool has changed."
@@ -164,7 +165,7 @@ jobs:
           delete-branch: true
           signoff: true
           base: main
-          token: ${{ secrets.GITHUB_TOKEN }}
+          token: ${{ secrets.WELAN_PAT }}
           labels: ${{ env.PR_LABEL }}
           reviewers: ${{ env.PR_REVIWER }}
 

--- a/test/scripts/update-kindnode-version.sh
+++ b/test/scripts/update-kindnode-version.sh
@@ -58,6 +58,11 @@ if [[ "$ORIGIN_K8S_VERSION" == "$KIND_NODE_VERSION" ]]; then
 elif [[ ${#KIND_NODE_VERSION} -gt ${#ORIGIN_K8S_VERSION} ]]; then
   echo "kind/node releases a new version, updates it in k8s matrix."
   sed -i 's/\'"${ORIGIN_K8S_VERSION}"/"${KIND_NODE_VERSION}"'/' $K8S_MATRIX_FILE_PATH
+  exit 0
+elif [[ ${#KIND_NODE_VERSION} -eq ${#ORIGIN_K8S_VERSION} && "$KIND_NODE_VERSION" != "$ORIGIN_K8S_VERSION" ]]; then
+  echo "kind/node released a new .z version, updated in the k8s matrix"
+  sed -i 's/\'"${ORIGIN_K8S_VERSION}"/"${KIND_NODE_VERSION}"'/' $K8S_MATRIX_FILE_PATH
+  exit 0
 else
   echo "update failed, please check."
   exit 1

